sqlglot/dialects/hive.py

```diff
@@ -272,13 +272,55 @@ class Parser(parser.Parser):
             "YEAR": lambda args: exp.Year(this=exp.TsOrDsToDate.from_arg_list(args)),
         }
 
+        FUNCTION_PARSERS = {
+            **parser.Parser.FUNCTION_PARSERS,
+            "TRANSFORM": lambda self: self._parse_query_transform(),
+        }
+
         PROPERTY_PARSERS = {
             **parser.Parser.PROPERTY_PARSERS,
             "WITH SERDEPROPERTIES": lambda self: exp.SerdeProperties(
                 expressions=self._parse_wrapped_csv(self._parse_property)
             ),
         }
 
+        def _parse_query_transform(self) -> exp.Anonymous | exp.QueryTransform:
+            args = self._parse_csv(self._parse_lambda)
+
+            index = self._index
+            self._match_r_paren()
+
+            row_format_before = self._parse_row_format(match_row=True)
+
+            record_writer = None
+            if self._match_text_seq("RECORDWRITER"):
+                record_writer = self._parse_string()
+
+            if not self._match(TokenType.USING):
+                self._retreat(index)
+                return self.expression(exp.Anonymous, this="TRANSFORM", expressions=args)
```

Comment:
i don't like using anonymous in the parser like this, just mkae transform a first class method

Reply:
Cool, will fix shortly